files/build_templates/per_namespace/bgp.service.j2

```diff
@@ -9,11 +9,12 @@ StartLimitIntervalSec=1200
 StartLimitBurst=3
 
 [Service]
-User={{ sonicadmin_user }}
```

Comment:
@jleveque @lguohan @yxieca For this directive `User=`, I changed it to `User=root` since Monit configuration file is only accessible by root. Do you have any concerns for this change?

Reply:
Please add the reason to the PR comments.

Reply:
Updated.

Reply:
better to make monit to work with sonicadmin_user, remove that restriction of monit.

Reply:
Will update.

Reply:
@lguohan @jleveque @yxieca I did the investigation to let Monit run by `sonicadmin_user`. 

The result is we can remove the restriction to let other users run Monit. However it will involve several modifications and I am not sure such change will introduce other issues or not:

1.We need change ownership of Monit control file `monitrc` from `root` to `admin`.  The reason is that Monit will check whether permission of Monit control file was `0700` or less `0700` when Monit was started to run. If the permission of control file was not, Monit will fail to start.

2.Once the ownership of Monit control file was changed from `root` to `admin`, we have to modify the init script of Monit `/etc/init.d/monit`. The reason is Monit will check whether the euid (effective uid ) of Monit process is euqal to uid of Monit control file. If we change the ownership of Monit control file from `root` to `admin`. The uid of this file will be `1000`, but the euid of Monit process is `0`. `0` represents the uid of root user.  why? The reason is that Monit process was started by `start-stop-daemon` in the init script of Monit `/etc/init.d/monitand`, it will inherit the uid of `start-stop-daemon`.  

In order to change the euid of Monit process to `1000`, we can add option `--chuid admin:admin` in the `start-stop-daemon` command to solve this issue.

3.We need change the location of Monit `pidfile`. Initially this file will be created in the directory `/run/monit.pid` .  After uid of Monit was changed from `0` to `1000`, it can not access `/run`.

3.We also need change the location of httpd socket file in Monit control file `monitrc`. Initially the file was created in the directory `/var/run/monit.sock` once Monit was started. After we change euid of Monit process from 0 to 1000, then Monit can not access the directory `/var/run`.

Reply:
Does this link help? https://serverfault.com/questions/50628/how-i-configure-monit-to-start-a-process-with-a-specific-user

Reply:
Thanks, Ying! This link introduced a method how should we start a process not as `root` in Monit configuration file. Currently what we want to do is to run Monit not as `root`.